### ISSUE TITLE
@JsonIgnore not working in 2.15.0 with transient fields having a public getter method

### ISSUE THREAD

**Issue by jayeshj100** [FasterXML/jackson-databind#3909]
**Describe the bug**
@JsonIgnore not working in 2.15.0 with transient fields having a public getter method. This seems to be a change behavior in 2.15.0. 

**Version information**
2.15.0

**To Reproduce**

```
public class Foo {
	private int a = 10;

	@JsonIgnore
	private transient int b = 11;

	public int getA() {
		return a;
	}

	public void setA(int a) {
		this.a = a;
	}

	public int getB() {
		return b;
	}

	public void setB(int b) {
		this.b = b;
	}

	public static void main(String args[]) throws JsonProcessingException {
		ObjectMapper ob = new ObjectMapper();
		System.out.println(ob.writeValueAsString(new Foo()));
	}
}
```
Output: {"a":10,"b":11}

**Expected behavior**
In an earlier version like 2.13.4, the output is: {"a":10}

**Additional context**
We noticed this behavior while trying to migrate from 2.13.4 to 2.15.0.

**Comment by yihtserns** [comment]
Seems to be duplicate of #3874?

**Comment by cowtowncoder** [comment]
Yes, duplicate.
While change to this usage was not the target, change was needed to fix #3682.

 Solution: instead of adding annotations on `transient` but instead on setter or getter; or, enable `MapperFeature.PROPAGATE_TRANSIENT_MARKER`.
